### ISSUE TITLE
Replace the NSFW directive with a more general-purpose metadata system

### ISSUE THREAD

**Issue by paavohuhtala** [TheBerkin/Rant#52]
The NSFW/SFW directives don't seem very useful outside of the standard dictionary. They could be removed and replaced with a system that allows you to attach metadata tags to dictionary entries and query those tags at run-time.

This would still allow tagging things as NSFW, but it would also enable things like tagging something as always plural (earnings, goods, clothes etc).

A sample (which is not a syntax proposal):
```
@class add tool
  >> spade/s
  | class blunt
  >> scissors/
  | class sharp
  | tag pair_of
@class remove tool

@class add orifice
  >> ear/s
  >> nostril/s
  @tag add nsfw
    >> arse/s
    >> dickhole/s
  @tag remove nsfw
@class remove orifice
```

```
Hey, is that your [hastag: tool1; pair_of]{pair of\s}<noun.singular-tool::=tool1>?
# Hey, is that your pair of scissors?
# Hey, is that your spade?

What if you'd stick it up your [hastag: orifice1; nsfw]{fucking\s}<noun.singular-orifice::=orifice1>?

This is a query for a rude orifice: <noun-singular-orifice#nsfw>.
```

**Comment by TheBerkin** [comment]
I'm having trouble understanding what the difference is, semantically, between tags and classes. Are tags supposed to be some kind of secondary class?

**Comment by cpancake** [comment]
I think the idea is that a tag is a sort of global class filter. So instead of NSFW being a class and having to filter out the NSFW class on every query, it would be filtered out by default.

**Comment by paavohuhtala** [comment]
I guess the difference between a class and a tag is somewhat arbitrary. The idea of a global class filter @cpancake mentioned is not something I originally thought of, but I could see it being useful.

If nothing else, I'd love to see [hastag:] or [hasclass:] being added.

**Comment by TheBerkin** [comment]
It would be useful to have global class filtering policies that can be turned on and off. That sounds like something that would be more of a header directive than something applied to individual entries.

The idea of having a `hasclass` function is interesting. Would this be functionally identical to a query's `-class` filter, except used to dynamically build queries?

**Comment by paavohuhtala** [comment]
Yes, pretty much.

**Comment by TheBerkin** [comment]
Good idea. That could eliminate the need for metapatterns in the next major version of Rant, which would make compiling a lot simpler. I'll keep this suggestion in mind for the future.

**Comment by TheBerkin** [comment]
Adapted version implemented as *hidden classes* in 6c41088114b2c1769cf45a3bbfa4e299043b25f1.
